### ISSUE TITLE
Adjust interface of join operator

### ISSUE THREAD

**Issue by mflueggen** [mweisgut/hyrise#1]

**Comment by mflueggen** [comment]
So far I am not very happy. I added an optional to the hash join to be able to consider additional predicates.
I am pretty sure we will have to extend the optimizer and add the LQPNodeType MultiPredicateJoin.

**Comment by mflueggen** [comment]
- [x] Think about interface
- [ ] Extend optimizer to detect and pass multi predicates to join (can be done without a working interface).
- [ ] Extend implementation of hash join to use additional predicates

**Comment by mweisgut** [comment]
> * [x]  Think about interface
> * [ ]  Extend optimizer to detect and pass multi predicates to join (can be done without a working interface).
> * [ ]  Extend implementation of hash join to use additional predicates

Based on our discussions with Stefan at our last weekly meeting, we do not need to extend the optimizer in the near future.
I would adapt your task list to the following:
- [x] Think about interface: we already did. We decided to just pass a vector (reference/pointer), that contains all the join predicates (column pairs and predicate conditions).
- [ ] add another constructor with the desired parameters
- [ ] adapt the implementation of the hash join so that the new constructor is the "main constructor" (the old constructor shall just call the new constructor). This step is just a temporary quick and dirty way not to adapt the calls of the old constructor in the entire code base.
- [ ] adapt all calls of the old constructor in the entire code base to be able to remove the old constructor
